app/src/main/java/com/cortado/minionify/preference/exceptions/PreferenceOperationFailure.java

```diff
@@ -0,0 +1,7 @@
+package com.cortado.minionify.preference.exceptions;
+
+public class PreferenceOperationFailure extends Exception {
```

Comment:
this is a bit misleading since you throw it when there is an illegal argument but from its name it sounds like an IO exception or something related to the prefs themselves. try to delete this exception.

Reply:
Removed